### PR TITLE
XP calculations in earnings calculator milestone display

### PR DESCRIPTION
**Pull request by vdaisson**
### All Submissions

* [X] Can you post a screenshot of your changes (if applicable)?
image

### New Feature Submissions

* [X] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?
 Issue #462

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

QoL change to show how much XP to claim to reach next milestone, based on current character's level and XP.

### CODE REVIEW

File: frontend/src/interfaces/Character.ts [hunk 1]
```diff
@@ -37,3 +37,29 @@ export function RequiredXp(level: number) {
   }
   return xp;
 }
+
+export function ExperienceTable() {
```
Comment (seiyria) [inline frontend/src/interfaces/Character.ts:41]: I would like to not have this hardcoded if possible. Ideally, there would be a way to export this from solidity. That's a @raymond-h question though. If it's not possible, he or I can merge this.

Comment (ricardoianelli) [reply frontend/src/interfaces/Character.ts:41]: If we're hardcoding that table I would suggest we save a table of experience per level. That way we can get rid of the iteration in the **getXpForNextMilestone** function also suggested in that PR. Would become something like that:

- ###  _frontend/src/interfaces/Character.ts_

> getXpForNextMilestone(level: number): number {
      const xpTable = ExperienceTable();
      const nextMilestoneLevel = this.getNextMilestoneLevel(level);
      return xpTable[nextMilestoneLevel] - this.currentCharacter.xp;
    },



- ### _frontend/src/components/smart/EarningsCalculator.vue_

> export function ExperienceTable() {
    return [
            0, 16, 17, 18, 19, 20, 22, 24, 26, 28, 30, 33, 36, 39, 42, 46, 50, 55, 60, 66, 
            72, 79, 86, 94, 103, 113, 124, 136, 149, 163, 178, 194, 211, 229, 248, 268, 289, 
            311, 334, 358, 383, 409, 436, 464, 493, 523, 554, 586, 619, 653, 688, 724, 761, 
            799, 838, 878, 919, 961, 1004, 1048, 1093, 1139, 1186, 1234, 1283, 1333, 1384, 
            1436, 1489, 1543, 1598, 1654, 1711, 1769, 1828, 1888, 1949, 2011, 2074, 2138, 2203, 
            2269, 2336, 2404, 2473, 2543, 2614, 2686, 2759, 2833, 2908, 2984, 3061, 3139, 3218, 
            3298, 3379, 3461, 3544, 3628, 3713, 3799, 3886, 3974, 4063, 4153, 4244, 4336, 4429, 
            4523, 4618, 4714, 4811, 4909, 5008, 5108, 5209, 5311, 5414, 5518, 5623, 5729, 5836, 
            5944, 6053, 6163, 6274, 6386, 6499, 6613, 6728, 6844, 6961, 7079, 7198, 7318, 7439, 
            7561, 7684, 7808, 7933, 8059, 8186, 8314, 8443, 8573, 8704, 8836, 8969, 9103, 9238, 
            9374, 9511, 9649, 9788, 9928, 10069, 10211, 10354, 10498, 10643, 10789, 10936, 11084, 
            11233, 11383, 11534, 11686, 11839, 11993, 12148, 12304, 12461, 12619, 12778, 12938, 
            13099, 13261, 13424, 13588, 13753, 13919, 14086, 14254, 14423, 14593, 14764, 14936, 
            15109, 15283, 15458, 15634, 15811, 15989, 16168, 16348, 16529, 16711, 16894, 17078, 
            17263, 17449, 17636, 17824, 18013, 18203, 18394, 18586, 18779, 18973, 19168, 19364, 
            19561, 19759, 19958, 20158, 20359, 20561, 20764, 20968, 21173, 21379, 21586, 21794, 
            22003, 22213, 22424, 22636, 22849, 23063, 23278, 23494, 23711, 23929, 24148, 24368, 
            24589, 24811, 25034, 25258, 25483, 25709, 25936, 26164, 26393, 26623, 26854, 27086, 
            27319, 27553, 27788, 28024, 28261, 28499, 28738, 28978
        ];
}


Still, I don't think returning the whole exp table on every function call is a good think to do, but I never studied neither javascript or typescript and don't know how to move that to a constant in a helper class.

Comment (ricardoianelli) [reply frontend/src/interfaces/Character.ts:41]: What do you think, @vdaisson ? Btw, great idea on this PR, that functionality would help a lot!

Comment (seiyria) [reply frontend/src/interfaces/Character.ts:41]: I don't think we need to copy paste anything, as stated. We can do the call once to get the table and call it good.

Comment (ricardoianelli) [reply frontend/src/interfaces/Character.ts:41]: Yup, if we can reuse instead of duplicating code, that would be better.

Comment (raymond-h) [reply frontend/src/interfaces/Character.ts:41]: There would need to be a function added to `characters.sol` to return the entire experience table in one call, but that is indeed doable.

**Pull request closed without merging**